src/deploy/NVA_build/noobaa_log_segregate.sh

```diff
@@ -82,39 +84,43 @@ while [ -n "$(ls -A "$log_dir")" ]; do
     log_file="$log_dir/$log_file.processing"
     # Iterate over each log records in the log file and do the following -
     # - Fetch the log bucket name from the log entry.
-    # - Fetch the log prefix from the log entry.
-    # - Create a file based on this log bucket name and log prefix if the file is not there.
+    # - Fetch the source bucket name from the log entry.
+    # - Create a file based on this log bucket name and source bucket name.
+    # - Check if the file is not there.
     # - append the current timestamp to the file name.
     # - Append the log entries to this file.
     # - Check the size of this file, if it is 50KB, move it to final location.
     # - Remove the log file once all log entries are processed.
     while read -r log; do
         IFS=, read -r -a log_fields <<< "$log"
         log_bucket_name=""
-        log_prefix_value=""
+        source_bucket_name=""
         bucket_log_file=""
         existing_log_file=""
         for log_field in "${log_fields[@]}"; do
             if [[ "$log_field" =~ "log_bucket" ]]; then
                 IFS=: read -r -a log_values <<< "$log_field"
                 log_bucket_name=$(echo "${log_values[1]}" | tr -d '"' | sed 's/^[ \t]*//; s/[ \t]*$//')
+                            continue
             fi
 
-            if [[ "$log_field" =~ "log_prefix" ]]; then
-                IFS=: read -r -a prefix_values <<< "$log_field"
-                log_prefix_value=$(echo "${prefix_values[1]}" | tr -d '"' | sed 's/^[ \t]*//; s/[ \t]*$//')
+            if [[ "$log_field" =~ "source_bucket" ]]; then
+                IFS=: read -r -a source_values <<< "$log_field"
+                source_bucket_name=$(echo "${source_values[1]}" | tr -d '"' | sed 's/^[ \t]*//; s/[ \t]*$//')
+                continue
             fi
```

Comment:
Why do we need to duplicate it?
can't it be a function?

Reply:
It could be a function. I modified the code and can see that it is working. However, I find the current code more readable and less prone to errors. Moreover, we are not saving any lines of code; all it is doing is adding more complexity to understand the code.